volatility3/framework/plugins/windows/strings.py

```diff
@@ -81,22 +87,45 @@ def _generator(self) -> Generator[Tuple, None, None]:
         last_prog: float = 0
         line_count: float = 0
         num_strings = len(string_list)
-        for offset, string in string_list:
+        for phys_offset, string in string_list:
             line_count += 1
-            try:
-                revmap_list = [
-                    name + ":" + hex(offset) for (name, offset) in revmap[offset >> 12]
-                ]
-            except (IndexError, KeyError):
-                revmap_list = ["FREE MEMORY"]
-            yield (
-                0,
-                (
-                    str(string, "latin-1"),
-                    format_hints.Hex(offset),
-                    ", ".join(revmap_list),
-                ),
+
+            # We should really take care of this in the revmap generator
+            mapping_entry = revmap.get(
+                phys_offset & 0xFFFFFFFFFFFFF000, [("In Unallocated Space", 0)]
             )
+            for item in mapping_entry:
+                region, offset = item
+                if "Process" in region:
+                    location = "Process"
+                    pid = region.split(" ")[1]
+                elif "kernel" in region:
+                    location = "Kernel"
+                    pid = -1
+                elif "In Unallocated Space" in region:
+                    location = "Unallocated Space"
+                    pid = -1
+                else:
+                    location = "Unknown"
+                    pid = -1
+
+                # Get the full virtual address not just the page start
+                if offset == 0:
+                    virtual_address = 0
+                else:
+                    virtual_address = (offset & ~0xFFF) | (phys_offset & 0xFFF)
```

Comment:
Again, these probably ought to be using the layer.mask value or something like that?